lighthouse-core/lib/client-rect-functions.js

```diff
@@ -0,0 +1,311 @@
+/**
+ * @license Copyright 2018 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @param {LH.Artifacts.ClientRect} cr
+ * @param {{x:number, y:number}} point
+ */
+/* istanbul ignore next */
+function rectContainsPoint(cr, {x, y}) {
+  return cr.left <= x && cr.right >= x && cr.top <= y && cr.bottom >= y;
+}
+
+/**
+ * @param {LH.Artifacts.ClientRect} cr1
+ * @param {LH.Artifacts.ClientRect} cr2
+ */
+/* istanbul ignore next */
+function rectContains(cr1, cr2) {
+  const topLeft = {
+    x: cr2.left,
+    y: cr2.top,
+  };
+  const topRight = {
+    x: cr2.right,
+    y: cr2.top,
+  };
+  const bottomLeft = {
+    x: cr2.left,
+    y: cr2.bottom,
+  };
+  const bottomRight = {
```

Comment:
maybe inline all these? Seems like `rectContains` will be a hot function, and most calls would fail on the first part of the below condition. Could be worth it to minimize object allocations